src/server/scripts/Spells/spell_warrior.cpp

```diff
@@ -1090,6 +1092,50 @@ class spell_warr_sword_and_board : public SpellScriptLoader
         }
 };
 
+// 215538 - Trauma
+// Updated 7.x
+class spell_warr_trauma : public SpellScriptLoader
+{
+public:
+    spell_warr_trauma() : SpellScriptLoader("spell_warr_trauma") { }
+
+    class spell_warr_trauma_AuraScript : public AuraScript
+    {
+        PrepareAuraScript(spell_warr_trauma_AuraScript);
+
+        bool Validate(SpellInfo const* /*spellInfo*/) override
+        {
+            return ValidateSpellInfo
+            ({
+                SPELL_WARRIOR_TRAUMA,
+                SPELL_WARRIOR_TRAUMA_EFFECT
+            });
+        }
+
+        void HandleProc(AuraEffect const* /*aurEff*/, ProcEventInfo& eventInfo)
+        {
+            PreventDefaultAction();
+            // Proc only with Slam & Whirlwind
+            if (!(eventInfo.GetDamageInfo()->GetSpellInfo()->SpellFamilyFlags[0] & 0x200000 || eventInfo.GetDamageInfo()->GetSpellInfo()->SpellFamilyFlags[1] & 0x4))
+                return;
+
+            //Get 25% of damage from the spell casted (Slam & Whirlwind)
+            int32 damage = eventInfo.GetDamageInfo()->GetDamage() * 25.0 / 100.0;
+            GetCaster()->CastCustomSpell(eventInfo.GetActionTarget(), SPELL_WARRIOR_TRAUMA_EFFECT, &damage, NULL, NULL, true);
```

Comment:
If target as has aura trauma (dot) and you proc again on this target. You need give a new amount of damage